azurelinuxagent/ga/cgroupapi.py

```diff
@@ -136,23 +158,106 @@ def get_systemd_run_commands(self):
+        The values returned can be None if the controller is not mounted or enabled.
+        """
+        pass    # pylint: disable=W0107
+
+    def get_process_cgroup_paths(self, process_id):
+        """
+        Cgroup version specific. Returns a tuple with the path of the cpu and memory cgroups for the given process.
+        The 'process_id' can be a numeric PID or the string "self" for the current process.
+        The values returned can be None if the process is not in a cgroup for that controller (e.g. the controller is
+        not mounted or enabled).
+        """
+        pass    # pylint: disable=W0107
+
+    def get_process_cgroup_relative_paths(self, process_id):  # pylint: disable=W0613
+        """
+        Cgroup version specific. Returns a tuple with the path of the cpu and memory cgroups for the given process
+        (relative to the mount point of the corresponding controller).
+        The 'process_id' can be a numeric PID or the string "self" for the current process.
+        The values returned can be None if the process is not in a cgroup for that controller (e.g. the controller is
+        not mounted).
         """
+        pass    # pylint: disable=W0107
+
+    def start_extension_command(self, extension_name, command, cmd_name, timeout, shell, cwd, env, stdout, stderr,
+                                error_code=ExtensionErrorCodes.PluginUnknownFailure):
+        """
+        Cgroup version specific. Starts extension command.
+        """
+        pass    # pylint: disable=W0107
+
+
+class SystemdCgroupsApiv1(SystemdCgroupsApi):
+    """
+    Cgroups v1 interface via systemd
+    """
+    def get_cgroup_mount_points(self):
         # the output of mount is similar to
-        #     $ mount -t cgroup
-        #     cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,name=systemd)
-        #     cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
-        #     cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
+        #     $ findmnt -t cgroup --noheadings
+        #     /sys/fs/cgroup/systemd          cgroup cgroup rw,nosuid,nodev,noexec,relatime,xattr,name=systemd
+        #     /sys/fs/cgroup/memory           cgroup cgroup rw,nosuid,nodev,noexec,relatime,memory
+        #     /sys/fs/cgroup/cpu,cpuacct      cgroup cgroup rw,nosuid,nodev,noexec,relatime,cpu,cpuacct
         #     etc
         #
-        if self._cgroup_mountpoints is None:
+        if not self._cgroup_mountpoints:
             cpu = None
             memory = None
-            for line in shellutil.run_command(['mount', '-t', 'cgroup']).splitlines():
-                match = re.search(r'on\s+(?P<path>/\S+(memory|cpuacct))\s', line)
+            for line in shellutil.run_command(['findmnt', '-t', 'cgroup', '--noheadings']).splitlines():
+                match = re.search(r'(?P<path>/\S+(memory|cpuacct))\s', line)
```

Comment:
on line 263 we should probably be looking for "cpu,cpuacct", rather than just "cpuacct"